content/250-postgres/500-backups.mdx

```diff
@@ -12,3 +12,133 @@ Please note that any database changes or events that occurred after the most rec
+
+To create backups, ensure you have the PostgreSQL command-line tools installed. Run the following commands based on your operating system:
+
+<TabbedContent code>
+
+<TabItem value="macOS">
+
+```terminal
+brew install postgresql@17
+which pg_dump
+which pg_restore
+```
+
+</TabItem>
+
+<TabItem value="Windows">
+
+```terminal
+# Download from the official PostgreSQL website:
+# https://www.postgresql.org/download/windows/
+# During installation, select "Command Line Tools".
+# Then verify with:
+
+where pg_dump
+where pg_restore
+```
+</TabItem>
+
+<TabItem value="Linux">
+
+```terminal
+sudo apt-get update
+sudo apt-get install postgresql-client-17
+which pg_dump
+which pg_restore
+```
+
+</TabItem>
+
+</TabbedContent>
+
+:::tip
+If you installed PostgreSQL but still see a “command not found” error for pg_dump or pg_restore, ensure your installation directory is in your system’s PATH environment variable.
+:::
+
+:::note
+Please make sure that you are installing Postgresql version 17. Other versions may cause errors during the backup process.
+:::
+
+### 1. Connecting to the database directly with `@prisma/ppg-tunnel`
+
+In your terminal, run `npx @prisma/ppg-tunnel` to establish a secure tunnel to your database.
+
+If you already have a `.env` file in your current directory with `DATABASE_URL` set, the tunnel CLI will automatically pick it up—no need to manually export it. However, if you haven't set up a `.env` file, you'll need to set the `DATABASE_URL` environment variable explicitly.
+
+To set environment variable (with your actual EA database URL):
+```bash
+export DATABASE_URL="prisma+postgres://accelerate.prisma-data.net/?api_key=eyJhbGciOiJIUzI..."
+```
+:::note
+If you explicitly set `DATABASE_URL` in your terminal, that value will take precedence over the one in your `.env` file.
+:::
+
+Run the tunnel:
+```bash
+npx @prisma/ppg-tunnel --host 127.0.0.1  --port 5432
+```
+
+You can provide any other port if you wish. A random port is assigned if no port value is provided.
+You should see output similar to:
+
+```cmd
+Prisma Postgres auth proxy listening on 127.0.0.1:5432 🚀
+
+Your connection is authenticated using your Prisma Postgres API key.
+...
+
+==============================
+hostname:  127.0.0.1
+port:      5432
+username:  <anything>
+password:  <none>
+==============================
+```
+
+:::note
+Please note that the port you will see in your output will be a randomly assigned port which may be different from the one mentioned here.
+Also, Keep this terminal window open so the tunnel remains active! If you close it, the tunnel disconnects.
+:::
+
+Copy the port number from the terminal output, you’ll need it in the next step for the `pg_dump` command.
+
+### 2. Creating the Backup with `pg_dump`
+
+With the tunnel running, you can now dump the database by running the following command:
+```bash
+PGSSLMODE=disable \
+pg_dump \
+  -h 127.0.0.1 \
+  -p 5432 \
```

Comment:
You mention that this could be different above, I think.

Reply:
Yes. But in the command we are specifying port 5432. I have added a note below that the port value should match what you get in output.